.github/workflows/ci.yml

```diff
@@ -19,172 +19,318 @@ jobs:
       - run: git remote add upstream https://github.com/modin-project/modin.git
       - run: git fetch upstream
       - run: npx commitlint --from upstream/master --to HEAD --verbose
+
   lint-black:
     name: lint (black)
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v2
         with:
           fetch-depth: 1
-      - uses: actions/setup-python@v1
+      - uses: actions/setup-python@v2
         with:
           python-version: "3.7.x"
           architecture: "x64"
       - run: pip install black
       - run: black --check --diff modin/
+
   lint-flake8:
     name: lint (flake8)
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v2
         with:
           fetch-depth: 1
-      - uses: actions/setup-python@v1
+      - uses: actions/setup-python@v2
         with:
           python-version: "3.7.x"
           architecture: "x64"
       - run: pip install flake8 flake8-print
       - run: flake8 --enable=T modin
+
   test-api:
     runs-on: ubuntu-latest
     name: test api
     steps:
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v2
         with:
           fetch-depth: 1
-      - uses: actions/setup-python@v1
+      - name: Cache conda
+        uses: actions/cache@v1
         with:
-          python-version: "3.7.x"
-          architecture: "x64"
+          path: ~/conda_pkgs_dir
+          # github.run_id using for create new cache entry for each run
```

Comment:
whose job is to clean the cache?

Reply:
Not really, the old cache will be deleted only if the total amount of caches stored under different keys exceeds the 5GB limit.

Reply:
What if we have so many concurrent CI runs that cache is out of space?..
And the question still stays - what is going to clean the cache?

Reply:
So - how many different variants of the cache (keys) we have - 12 (2 - OS, 3 - python versions, 2 managers - pip and conda). At the moment, one cache write for pep is <50MB, one cache for conda is <300MB. Less than 3GB in total for one CI run. So part of cache will be clean up by 2 concurrent CI runs.

Reply:
I may not use the Conda cache, only pip. Then problems should not arise up to 10 parallel CI runs. Need to look at performance.

Reply:
Let's keep in mind this restriction then, and use cache for all runs for now, as I don't think we have enough resources for a lot of concurrent CI runs anyway.

Maybe dropping a comment about that in the `.yml` file would be good.

Reply:
Removing the conda cache doesn't make performance worse. So let's cache only pip packages.